src/pages/lesson-details/LessonDetails.tsx

```diff
@@ -78,6 +89,50 @@ const LessonDetails = () => {
     })
   }
 
+  const getCooperation = useCallback(() => {
+    return cooperationService.getCooperationById(id)
+  }, [id])
+
+  const { data: cooperation } = useQuery({
+    queryFn: getCooperation,
+    queryKey: ['cooperation', id]
+  })
+
+  useEffect(() => {
+    if (!cooperation?.sections) return
+
+    cooperation?.sections?.some((section) => {
+      const resource = section.resources?.find(
+        (resource) => resource.resource._id === lessonId
+      )
+
+      if (resource) {
+        setCompletionStatus(resource.completionStatus)
+        return true
+      }
+    })
+  }, [cooperation, lessonId, setCompletionStatus])
+
+  const handleUpdateLessonStatus = useCallback(() => {
+    return cooperationService.updateResourceCompletionStatus({
+      id,
+      resourceId: lessonId,
+      completionStatus:
+        completionStatus === CompletionStatusEnum.Active
+          ? CompletionStatusEnum.Completed
+          : completionStatus
+    })
+  }, [id, lessonId, completionStatus])
+
+  const handleResponse = () => {
+    setCompletionStatus(CompletionStatusEnum.Completed)
+  }
+
+  const { mutate: updateLessonStatus } = useMutation({
+    mutationFn: handleUpdateLessonStatus,
```

Comment:
Please, rewrite it. You should call `mutate` function of the `useMutation` hook inside `useCallback` with params that you need. It will help you to just pass this callback to the `onClick` handler of the button without additional wrappers


```suggestion
    mutationFn: cooperationService.updateResourceCompletionStatus,
```

Reply:
done